### ISSUE TITLE
No error thrown when calling `concat` with `T[]` and `T[][]`

### ISSUE THREAD

**Issue by thomasschafer** [microsoft/TypeScript#55844]
### 🔎 Search Terms

"concat", "concat wrong type", "concat no error"

### 🕗 Version & Regression Information

- This is a bug
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `concat`

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=3.3.3#code/MYewdgzgLgBAhgJwQRhgXhgbU8gurgKFElkQQCZ14lkA6Y4OKACk3NwEojwIQAbAKa0+IAObMy5LtxLUEAZirYARHGX4Z0OQBYqZefXCMWmZQCN1XYr0HCxEpNo5A

### 💻 Code

```ts
const arr1 = [[1]]
const arr2 = arr1.concat([2]) // No error

const arr3 = [["a"]]
const arr4 = arr3.concat(["b"]) // No error
```

### 🙁 Actual behavior

No error is shown, and type of the result is incorrectly inferred as `T[]`

### 🙂 Expected behavior

Either an error should be shown, or the result of `concat` should be `(T | T[])[]`

**Comment by MartinJohns** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/36554#issuecomment-580921643.
